tests/test_backend_equivalence.py

```diff
@@ -72,34 +73,80 @@ def pandas_impls():
     return {name: PandasTableImpl(name, df) for name, df in dataframes.items()}
 
 
-def sql_impls():
-    engine = sqlalchemy.create_engine("sqlite:///:memory:")
+def sql_conn_to_impls(conn: str, project_id=None, dataset=None):
+    engine = sqlalchemy.create_engine(conn)
     impls = {}
+    print("THIS IS VERY EXPENSIVE")
     for name, df in dataframes.items():
-        df.to_sql(name, engine, index=False, if_exists="replace")
-        impls[name] = SQLTableImpl(engine, name)
+        if engine.dialect.name == "bigquery":
+            if not (project_id and dataset):
+                raise ValueError(
+                    "Project Id and dataset names are required for bigquery"
+                )
+            # name = f"{dataset}.{name}"
+            impls[name] = SQLTableImpl(engine, f"{dataset}.{name}")
+        else:
+            df.to_sql(name, engine, index=False, if_exists="replace")
+            impls[name] = SQLTableImpl(engine, name)
     return impls
 
 
-impls = {"pandas": pandas_impls, "sql_sqlite": sql_impls}
+def sqlite_impls():
+    return sql_conn_to_impls("sqlite:///:memory:")
+
+
+# def mssql_impls():
+#     user = "sa"
+#     password = "QuantCompare37"
+#     localhost = "127.0.0.1"
+#     db_name = "master"
+#     local_conn = f"mssql+pyodbc://{user}:{password}@{localhost}:1433/{db_name}?driver=ODBC+Driver+18+for+SQL+Server&encrypt=no"
+#     return sql_conn_to_impls(local_conn)
+
+
+def postgresql_impls():
+    user = "sa"
+    password = "Pydiverse23"
+    local_conn = f"postgresql://{user}:{password}@localhost:5432/"
+    return sql_conn_to_impls(local_conn)
+
+
+# def bigquery_impls():
+#     dataset = "dataset_test"
+#     project_id = "project-id"
+#     auth_path = "bq_key.json"
+#     credentials = service_account.Credentials.from_service_account_file(auth_path)
+#     local_conn = (  # ?DataSetId={dataset}&ProjectId={project_id}&InitiateOAuth=GETANDREFRESH&OAuthSettingsLocation={auth_path}"
+#         f"bigquery://{project_id}"
+#     )
+#     return sql_conn_to_impls(local_conn, project_id=project_id, dataset=dataset)
```

Comment:
let's keep this code commented here until we decided how to setup cloud databases in github actions